Error when restoring a backup with cypher-shell

Hello!

Today, I was trying to restore a backup in my local computer (dev) from the dev environment. So first, I did the backup with the following line:
```
CALL apoc.export.cypher.all('backup.cypher',{format:'cypher-shell', useOptimizations: {type: "UNWIND_BATCH", unwindBatchSize: 20}})
```

Then in my local computer, I open the powershell (yeah, Win10) and went to the project folder and typed:
```
type .\import\backup.cypher | .\bin\cypher-shell.bat  -u neo4j -p password -d database_name --format plain
```

## Expected Behavior (Mandatory)

I was hoping to get the backup on my computer, but then the process failed.

## Actual Behavior (Mandatory)
I got the next error:

**An error occurred while in an open transaction. The transaction will be rolled back and terminated. Error: Tried to execute Schema modification after executing Write query**

After running with --debug, I get the following error:

```
org.neo4j.shell.state.ErrorWhileInTransactionException: An error occurred while in an open transaction. The transaction will be rolled back and terminated. Error: Tried to ex
ecute Schema modification after executing Write query
        at org.neo4j.shell.state.BoltStateHandler.handleException(BoltStateHandler.java:215)
        at org.neo4j.shell.CypherShell.executeCypher(CypherShell.java:147)
        at org.neo4j.shell.CypherShell.execute(CypherShell.java:113)
        at org.neo4j.shell.cli.NonInteractiveShellRunner.runUntilEnd(NonInteractiveShellRunner.java:91)
        at org.neo4j.shell.Main.runShell(Main.java:165)
        at org.neo4j.shell.Main.startShell(Main.java:118)
        at org.neo4j.shell.Main.main(Main.java:88)
Caused by: org.neo4j.driver.exceptions.ClientException: Tried to execute Schema modification after executing Write query
        at org.neo4j.driver.internal.util.Futures.blockingGet(Futures.java:143)
        at org.neo4j.driver.internal.InternalTransaction.run(InternalTransaction.java:60)
        at org.neo4j.shell.state.BoltStateHandler.getBoltResult(BoltStateHandler.java:559)
        at org.neo4j.shell.state.BoltStateHandler.runCypher(BoltStateHandler.java:451)
        at org.neo4j.shell.CypherShell.executeCypher(CypherShell.java:136)
        ... 5 more
        Suppressed: org.neo4j.driver.internal.util.ErrorUtil$InternalExceptionCause
                at org.neo4j.driver.internal.util.ErrorUtil.newNeo4jError(ErrorUtil.java:85)
                at org.neo4j.driver.internal.async.inbound.InboundMessageDispatcher.handleFailureMessage(InboundMessageDispatcher.java:119)
                at org.neo4j.driver.internal.messaging.common.CommonMessageReader.unpackFailureMessage(CommonMessageReader.java:83)
                at org.neo4j.driver.internal.messaging.common.CommonMessageReader.read(CommonMessageReader.java:59)
                at org.neo4j.driver.internal.async.inbound.InboundMessageHandler.channelRead0(InboundMessageHandler.java:83)
                at org.neo4j.driver.internal.async.inbound.InboundMessageHandler.channelRead0(InboundMessageHandler.java:35)
                at org.neo4j.driver.internal.shaded.io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
                at org.neo4j.driver.internal.shaded.io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
                at org.neo4j.driver.internal.shaded.io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
                at org.neo4j.driver.internal.async.inbound.MessageDecoder.channelRead(MessageDecoder.java:47)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
                at org.neo4j.driver.internal.shaded.io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
                at org.neo4j.driver.internal.shaded.io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:311)
                at org.neo4j.driver.internal.shaded.io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:432)
                at org.neo4j.driver.internal.shaded.io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
                at org.neo4j.driver.internal.shaded.io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
                at org.neo4j.driver.internal.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
                at org.neo4j.driver.internal.shaded.io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
                at org.neo4j.driver.internal.shaded.io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
                at org.neo4j.driver.internal.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719)
                at org.neo4j.driver.internal.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
                at org.neo4j.driver.internal.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
                at org.neo4j.driver.internal.shaded.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
                at org.neo4j.driver.internal.shaded.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
                at org.neo4j.driver.internal.shaded.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
                at org.neo4j.driver.internal.shaded.io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
                at java.base/java.lang.Thread.run(Thread.java:834)
```

## Specifications 

Server is on linux running Neo4j 4.3.3 with APOC 4.3.0.1
Local computer is with windows with Neo4j 4.3.3 with APOC 4.3.0.4

I haver a backup from like 2 moths ago and works fine. Also, we just started to use the lucerne index, I do not know if that might be a problem.

Thanks!

--- a/core/src/main/java/apoc/export/cypher/formatter/AbstractCypherFormatter.java
+++ b/core/src/main/java/apoc/export/cypher/formatter/AbstractCypherFormatter.java
@@ -33,9 +33,8 @@ abstract class AbstractCypherFormatter implements CypherFormatter {
 
 	private static final String STATEMENT_CONSTRAINTS = "CREATE CONSTRAINT ON (node:%s) ASSERT (%s) %s;";
 
-	private static final String STATEMENT_NODE_FULLTEXT_IDX = "CALL db.index.fulltext.createNodeIndex('%s',[%s],[%s]);";
-	private static final String STATEMENT_REL_FULLTEXT_IDX = "CALL db.index.fulltext.createRelationshipIndex('%s',[%s],[%s]);";
-	public static final String PROPERTY_QUOTING_FORMAT = "'%s'";
+	private static final String STATEMENT_NODE_FULLTEXT_IDX = "CREATE FULLTEXT INDEX %s FOR (n:%s) ON EACH [%s];";
+	private static final String STATEMENT_REL_FULLTEXT_IDX = "CREATE FULLTEXT INDEX %s FOR ()-[rel:%s]-() ON EACH [%s];";
 
 	@Override
 	public String statementForCleanUp(int batchSize) {
@@ -59,12 +58,9 @@ abstract class AbstractCypherFormatter implements CypherFormatter {
 		String label = StreamSupport.stream(labels.spliterator(), false)
 				.map(Label::name)
 				.map(Util::quote)
-				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
-				.collect(Collectors.joining(","));
-		String key = StreamSupport.stream(keys.spliterator(), false)
-				.map(Util::quote)
-				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
-				.collect(Collectors.joining(","));
+				.collect(Collectors.joining("|"));
+		String key = getPropertiesQuoted(keys, "n.");
+
 		return String.format(STATEMENT_NODE_FULLTEXT_IDX, name, label, key);
 	}
 
@@ -73,12 +69,9 @@ abstract class AbstractCypherFormatter implements CypherFormatter {
 		String type = StreamSupport.stream(types.spliterator(), false)
 				.map(RelationshipType::name)
 				.map(Util::quote)
-				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
-				.collect(Collectors.joining(","));
-		String key = StreamSupport.stream(keys.spliterator(), false)
-				.map(Util::quote)
-				.map(s -> String.format(PROPERTY_QUOTING_FORMAT, s))
-				.collect(Collectors.joining(","));
+				.collect(Collectors.joining("|"));
+		String key = getPropertiesQuoted(keys, "rel.");
+		
 		return String.format(STATEMENT_REL_FULLTEXT_IDX, name, type, key);
 	}
 

--- a/core/src/test/java/apoc/export/cypher/ExportCypherTest.java
+++ b/core/src/test/java/apoc/export/cypher/ExportCypherTest.java
@@ -555,20 +555,20 @@ public class ExportCypherTest {
     @Test
     public void exportMultiTokenIndex() {
         // given
-        db.executeTransactionally("CREATE (n:TempNode {value:'value'})");
-        db.executeTransactionally("CREATE (n:TempNode2 {value:'value'})");
-        db.executeTransactionally("CALL db.index.fulltext.createNodeIndex('MyCoolNodeFulltextIndex',['TempNode', 'TempNode2'],['value'])");
+        db.executeTransactionally("CREATE (n:TempNode {value:'value', value2:'value'})");
+        db.executeTransactionally("CREATE (n:TempNode2 {value:'value', value:'value2'})");
+        db.executeTransactionally("CALL db.index.fulltext.createNodeIndex('MyCoolNodeFulltextIndex',['TempNode', 'TempNode2'],['value', 'value2'])");
 
         String query = "MATCH (t:TempNode) return t";
         String file = null;
         Map<String, Object> config = map("awaitForIndexes", 3000);
         String expected = String.format(":begin%n" +
-                "CALL db.index.fulltext.createNodeIndex('MyCoolNodeFulltextIndex',['TempNode','TempNode2'],['value']);%n" +
+                "CREATE FULLTEXT INDEX MyCoolNodeFulltextIndex FOR (n:TempNode|TempNode2) ON EACH [n.value, n.value2];%n" +
                 "CREATE CONSTRAINT ON (node:`UNIQUE IMPORT LABEL`) ASSERT (node.`UNIQUE IMPORT ID`) IS UNIQUE;%n" +
                 ":commit%n" +
                 "CALL db.awaitIndexes(3000);%n" +
                 ":begin%n" +
-                "UNWIND [{_id:3, properties:{value:\"value\"}}] AS row%n" +
+                "UNWIND [{_id:3, properties:{value2:\"value\", value:\"value\"}}] AS row%n" +
                 "CREATE (n:`UNIQUE IMPORT LABEL`{`UNIQUE IMPORT ID`: row._id}) SET n += row.properties SET n:TempNode;%n" +
                 ":commit%n" +
                 ":begin%n" +

